app/controllers/application_controller.rb

```diff
@@ -419,14 +419,22 @@ def use_caching?
   def check_user_status
     if current_user.is_a?(User) && (current_user.suspended? || current_user.banned?)
       if current_user.suspended?
-        flash[:error] = t('suspension_notice', default: "Your account has been suspended until %{suspended_until}. You may not add or edit content until your suspension has been resolved. Please <a href=\"#{new_abuse_report_path}\">contact Abuse</a> for more information.", suspended_until: localize(current_user.suspended_until)).html_safe
+        flash[:error] = t("suspension_notice_html", default: "Your account has been suspended until %{suspended_until}. You may not add or edit content until your suspension has been resolved. Please <a href=\"#{new_abuse_report_path}\">contact Abuse</a> for more information.", suspended_until: localize(current_user.suspended_until))
       else
-        flash[:error] = t('ban_notice', default: "Your account has been banned. You are not permitted to add or edit archive content. Please <a href=\"#{new_abuse_report_path}\">contact Abuse</a> for more information.").html_safe
+        flash[:error] = t("ban_notice_html", default: "Your account has been banned. You are not permitted to add or edit archive content. Please <a href=\"#{new_abuse_report_path}\">contact Abuse</a> for more information.").html_safe
```

Comment:
Using `_html` in keys to avoid calls to `html_safe` is good, but unfortunately in Rails 6.1 it only works in views, not controllers. Rails 7.0 [introduces it for controllers as well](https://github.com/rails/rails/commit/a5247bb90895f843c45fb095a4072e8ef30eaa4e), but it'll probably be at least 6 months before that upgrade happens (judging by when the Rails upgrades happened in previous years).